src/main/java/org/broadinstitute/hellbender/utils/diffengine/DiffEngine.java

```diff
@@ -0,0 +1,369 @@
+        } else {
+            final String[] parts = Arrays.copyOf(partsArg, partsArg.length);
+            for ( int i = 0; i < stop; i++ ) {
+                parts[i] = "*";
+            }
+            return String.join(".", parts);
+        }
+    }
+
+
+    /**
+     * Returns an unmodifiable map of reader names to readers.
+     */
+    Map<String, DiffableReader> getReaders() {
+        return Collections.unmodifiableMap(readers);
+    }
+
+    DiffableReader getReader(final String name) {
+        return readers.get(name);
+    }
+
+    /**
+     * Returns a reader appropriate for this file, or null if no such reader exists
+     * @param file
+     * @return
+     */
+    DiffableReader findReaderForFile(final File file) {
+        for ( final DiffableReader reader : readers.values() ) {
+            if (reader.canRead(file)) {
+                return reader;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns true if reader appropriate for this file, or false if no such reader exists
+     * @param file
+     * @return
+     */
+    public boolean canRead(final File file) {
+        return findReaderForFile(file) != null;
+    }
+
+    DiffElement createDiffableFromFile(final File file, final int maxElementsToRead) throws IOException {
+        final DiffableReader reader = findReaderForFile(file);
+        if ( reader == null )
+            throw new UserException("Unsupported file type: " + file);
+        else
+            return reader.readFromFile(file, maxElementsToRead);
+    }
+
+    private static boolean simpleDiffFiles(final File masterFile, final File testFile, final int maxElementsToRead, final DiffEngine.SummaryReportParams params) throws IOException {
+        final DiffEngine diffEngine = new DiffEngine();
+
+        if ( diffEngine.canRead(masterFile) && diffEngine.canRead(testFile) ) {
+            final DiffElement master = diffEngine.createDiffableFromFile(masterFile, maxElementsToRead);
+            final DiffElement test = diffEngine.createDiffableFromFile(testFile, maxElementsToRead);
+            final List<Difference> diffs = master.diffElements(test);
+            DiffEngine.reportSummarizedDifferences(diffs, params);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public String diffFiles(final File masterFile, final File testFile) throws IOException {
+        String diffEngineOutput = "";
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try(final PrintStream ps = new PrintStream(baos)) {
+            final DiffEngine.SummaryReportParams params = new DiffEngine.SummaryReportParams(ps, 20, 10, 0, MAX_RAW_DIFFS_TO_SUMMARIZE, false);
+            final boolean success = DiffEngine.simpleDiffFiles(masterFile, testFile, MAX_RECORDS_TO_READ, params);
+            if (success) {
+                diffEngineOutput = baos.toString();
+                BaseTest.log(diffEngineOutput);
```

Comment:
This (officially) non-test class should not be calling into a `BaseTest` method.